vhdbuilder/packer/vhd-image-builder-cvm.json

```diff
@@ -0,0 +1,669 @@
+{
+  "variables": {
+    "subscription_id": "{{env `AZURE_SUBSCRIPTION_ID`}}",
+    "location": "{{env `PACKER_BUILD_LOCATION`}}",
+    "vm_size": "{{env `AZURE_VM_SIZE`}}",
+    "build_definition_name": "{{env `BUILD_DEFINITION_NAME`}}",
+    "build_number": "{{env `BUILD_NUMBER`}}",
+    "build_id": "{{env `BUILD_ID`}}",
+    "commit": "{{env `GIT_VERSION`}}",
+    "feature_flags": "{{env `FEATURE_FLAGS`}}",
+    "image_version": "{{env `IMAGE_VERSION`}}",
+    "os_version": "{{env `OS_VERSION`}}",
+    "sku_name": "{{env `SKU_NAME`}}",
+    "hyperv_generation": "{{env `HYPERV_GENERATION`}}",
+    "sig_gallery_name": "{{env `SIG_GALLERY_NAME`}}",
+    "sig_image_name": "{{env `SIG_IMAGE_NAME`}}",
+    "sig_image_version": "{{env `SIG_IMAGE_VERSION`}}",
+    "container_runtime": "{{env `CONTAINER_RUNTIME`}}",
+    "teleportd_plugin_download_url": "{{env `TELEPORTD_PLUGIN_DOWNLOAD_URL`}}",
+    "captured_sig_version": "{{env `${CAPTURED_SIG_VERSION`}}",
+    "enable_fips": "{{env `ENABLE_FIPS`}}",
+    "img_publisher": "{{env `IMG_PUBLISHER`}}",
+    "img_offer": "{{env `IMG_OFFER`}}",
+    "img_sku": "{{env `IMG_SKU`}}",
+    "img_version": "{{env `IMG_VERSION`}}",
+    "sgx_install": "{{env `SGX_INSTALL`}}",
+    "vnet_resource_group_name": "{{env `VNET_RESOURCE_GROUP_NAME`}}",
+    "vnet_name": "{{env `VNET_NAME`}}",
+    "subnet_name": "{{env `SUBNET_NAME`}}",
+    "private_packages_url": "{{env `PRIVATE_PACKAGES_URL`}}",
+    "branch": "{{env `BRANCH`}}",
+    "vhd_build_timestamp": "{{user `VHD_BUILD_TIMESTAMP`}}"
+  },
+  "builders": [
+    {
+      "type": "azure-arm",
+      "subscription_id": "{{user `subscription_id`}}",
+      "virtual_network_resource_group_name": "{{user `vnet_resource_group_name`}}",
+      "virtual_network_name": "{{user `vnet_name`}}",
+      "virtual_network_subnet_name": "{{user `subnet_name`}}",
+      "ssh_read_write_timeout": "5m",
+      "os_type": "Linux",
+      "os_disk_size_gb": 30,
+      "image_publisher": "{{user `img_publisher`}}",
+      "image_offer": "{{user `img_offer`}}",
+      "image_sku": "{{user `img_sku`}}",
+      "image_version": "{{user `img_version`}}",
+      "azure_tags": {
+        "buildDefinitionName": "{{user `build_definition_name`}}",
+        "buildNumber": "{{user `build_number`}}",
+        "buildId": "{{user `build_id`}}",
+        "SkipLinuxAzSecPack": "true",
+        "os": "Linux",
+        "now": "{{user `create_time`}}",
+        "createdBy": "aks-vhd-pipeline",
+        "image_sku": "{{user `img_sku`}}",
+        "branch": "{{user `branch`}}"
+      },
+      "location": "{{user `location`}}",
+      "vm_size": "{{user `vm_size`}}",
+      "polling_duration_timeout": "1h",
+      "managed_image_storage_account_type": "Premium_LRS",
+      "secure_boot_enabled": true,
+      "vtpm_enabled": true,
+      "security_type": "ConfidentialVM",
```

Comment:
are these fields the only diff with the existing base template? if so we might be able to reuse the base template and pass these values as variables initialized within init-variables.sh

Reply:
These fields and the two in the Shared Image Gallery Destination block. Additionally, the CVM template does not have the managed-image name/RG variables, as the CVM can't have a managed-image source.